casper/src/main/scala/coop/rchain/casper/genesis/Genesis.scala

```diff
@@ -23,15 +31,29 @@ final case class Genesis(
 )
 
 object Genesis {
+  val genesisPubKey = PublicKey(Array[Byte]())
 
-  val NonNegativeMergeableTagName: Par = {
-    val rand = Tools.unforgeableNameRng(
-      StandardDeploys.nonNegativeNumberPubKey,
-      StandardDeploys.nonNegativeNumberTimestamp
-    )
-    import coop.rchain.models.rholang.implicits._
+  // using fixed 0 to make sure unforgeable name is predictable without configuring blockNumber
+  val genesisRandomSeedBlockNumber = 0L
+
+  def nonNegativeMergeableTagName(
+      shardId: String
+  ): Par = {
+    // NonNegative contract is the 4th contract deployed in the genesis, start from 0. Index should be 3
+    val nonNegativeContractIndex: Byte = 3
+    val rand = BlockRandomSeed
+      .generateSplitRandomNumber(
+        BlockRandomSeed(
+          shardId,
+          genesisRandomSeedBlockNumber,
+          genesisPubKey, // using fixed pubkey to make sure unforgeable name is predictable without configuring sender
+          Blake2b256Hash.fromByteString(emptyStateHashFixed)
```

Comment:
For genesis all fields can be empty except shard ID. In that case we don't need to track empty state hash to be compatible for specific network.